datashare-index/src/main/java/org/icij/datashare/text/indexing/elasticsearch/SourceExtractor.java

```diff
@@ -48,27 +52,37 @@ public InputStream getSource(final Project project, final Document document) thr
             }
         } else {
             LOGGER.info("extracting embedded document " + Identifier.shorten(document.getId(), 4) + " from root document " + document.getPath());
-            TikaDocumentSource source;
-            EmbeddedDocumentMemoryExtractor embeddedExtractor;
-            DigestIdentifier identifier;
-            if (document.getId().length() == SHA_384.digestLength) {
-                embeddedExtractor = new EmbeddedDocumentMemoryExtractor(new UpdatableDigester(project.getId(), SHA_384.toString()));
-                identifier = new DigestIdentifier(SHA_384.toString(), Charset.defaultCharset());
-            } else {
-                // backward compatibility
-                Hasher hasher = Hasher.valueOf(document.getId().length());
-                embeddedExtractor = new EmbeddedDocumentMemoryExtractor(
-                        new CommonsDigester(20 * 1024 * 1024, hasher.toString().replace("-", "")), hasher.toString(), false);
-                identifier = new DigestIdentifier(hasher.toString(), Charset.defaultCharset());
-            }
+            return getEmbeddedSource(project, document);
+        }
+    }
+
+    public InputStream getEmbeddedSource(final Project project, final Document document) throws FileNotFoundException {
+        Hasher hasher = Hasher.valueOf(document.getId().length());
+        String algorithm = hasher.toString();
+        List<DigestingParser.Digester> digesters = new ArrayList<>(List.of());
+        digesters.add(new CommonsDigester(20 * 1024 * 1024, algorithm.replace("-", "")));
+        digesters.add(new UpdatableDigester(project.getId(), algorithm));
+
+        for (DigestingParser.Digester digester : digesters) {
```

Comment:
```suggestion
       // Try each digester to find embedded doc
        for (DigestingParser.Digester digester : digesters) {
```
I would have a comment to "justify" the change